examples/.test/internal-types-export/src/server.ts

```diff
@@ -3,9 +3,20 @@ import { initTRPC } from '@trpc/server';
 // needs to be exported for the test to be valid
 export const t = initTRPC.create();
 
+const someMiddleware = t.middleware(({ next }) => {
+  return next();
+});
+
+export function genericRouter<S extends (value: any) => unknown>(schema: S) {
+  return t.router({
+    foo: t.procedure.output(schema).query(() => 'bar'),
```

Comment:
Soooo, we've just noticed that this is incorrect code which showed up while doing work on next major

You cannot know that `'bar'` is correct when "before" the `TSchema` is set

so just FYI @milohansen, this will break, unless you typecast `'bar'`, when you upgrade to v11

Reply:
See #5330 for what you should be doing instead